src/activities/categorization/Categorization.qml

```diff
@@ -123,73 +122,73 @@ ActivityBase {
                         onCheckedChanged: {
-                            if(easyModeBox.checked) {
-                                items.mode = "easy"
-                                menuScreen.iAmReady.visible = false
-                            }
+                            items.instructionsVisible = instructionsBox.checked
+                            menuScreen.iAmReady.visible = ((!items.scoreVisible && !items.instructionsVisible) || (!items.instructionsVisible && items.scoreVisible))
                         }
                     }
 
                     GCDialogCheckBox {
-                        id: mediumModeBox
-                        width: easyModeBox.width
-                        text: qsTr("Instructions visible and score invisible")
-                        checked: (items.mode == "medium") ? true : false
-                        exclusiveGroup: configOptions
+                        id: scoreBox
+                        width: instructionsBox.width
+                        text: qsTr("Score visible")
+                        checked: items.scoreVisible
                         onCheckedChanged: {
-                            if(mediumModeBox.checked) {
-                                items.mode = "medium"
-                                menuScreen.iAmReady.visible = false
-                            }
+                            items.scoreVisible = scoreBox.checked
+                            menuScreen.iAmReady.visible = ((!items.scoreVisible && !items.instructionsVisible) || (!items.instructionsVisible && items.scoreVisible))
                         }
                     }
 
                     GCDialogCheckBox {
-                        id: expertModeBox
-                        width: easyModeBox.width
-                        text: qsTr("Instructions and score invisible")
-                        checked: (items.mode == "expert") ? true : false
-                        exclusiveGroup: configOptions
+                        id: categoryImageBox
+                        width: instructionsBox.width
+                        text: qsTr("Category image visible")
+                        checked: items.categoryImageChecked
                         onCheckedChanged: {
-                            if(expertModeBox.checked) {
-                                items.mode = "expert"
-                                menuScreen.iAmReady.visible = true
-                            }
+                            items.categoryImageVisible = categoryImageBox.checked
+                            menuScreen.iAmReady.visible = ((!items.scoreVisible && !items.instructionsVisible) || (!items.instructionsVisible && items.scoreVisible))
+
                         }
                     }
                 }
             }
             onLoadData: {
-                if(dataToSave && dataToSave["mode"])
-                    items.mode = dataToSave["mode"]
+                if(dataToSave) {
```

Comment:
you need to at least add a: && dataToSave["scoreVisible"]
else it will fail at first access when you don't have it already stored